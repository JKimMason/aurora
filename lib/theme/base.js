export default {
  colors: {
    // Main colors
    primary: "#3498db",
    success: "#2ecc71",
    danger: "#e74c3c",

    // Background
    background: "#F3F3F3",
    secondaryBackground: "white",

    // Lines and borders
    border: "#E5E5E5",

    // Text
    lightText: "#777777",
    text: "#282828"
  },

  font: "Helvetica",

  spacing: {
    padding: "1em"
  }
};
