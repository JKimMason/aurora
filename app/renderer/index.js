import React from "react";
import ReactDOM from "react-dom";
import Electron from "./components/Electron.js";
import Core from "../../lib/core";
import { Store } from "../../lib/mutate";

const render = Page => {
  ReactDOM.render(
    <Electron>
      <Page />
    </Electron>,
    document.getElementById("root")
  );
};

export default {
  index: () => {
    render(Core);
  },
  mutations: () => {
    render(Store);
  }
};
