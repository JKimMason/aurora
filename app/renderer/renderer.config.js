/**
 * Webpack config for compiling renderer
 */

const path = require("path");

module.exports = {
  context: path.resolve(__dirname),
  entry: ["babel-polyfill", "./index.js"],
  output: {
    path: path.resolve("app/.compiled"),
    filename: "renderer.bundle.js",
    library: "app",
    libraryTarget: "commonjs2"
  },
  target: "electron-renderer",
  module: {
    loaders: [
      {
        test: /\.js$/,
        loader: "babel-loader",
        exclude: /node_modules/,
        options: {
          presets: ["env", "react", "stage-2"]
        }
      },
      {
        test: /\.jsx$/,
        loader: "babel-loader",
        exclude: /node_modules/,
        options: {
          presets: ["env", "react", "stage-2"]
        }
      }
    ]
  }
};
