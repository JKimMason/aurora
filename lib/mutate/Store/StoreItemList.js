import React from "react";
import StoreItem from "./StoreItem";
import { removePrefix } from "./util";
import PropTypes from "prop-types";

const StoreItemList = ({ items, onClick }) => {
  const List = items.map(item => {
    const title = removePrefix(item.package.name);

    // Handle the case where something is named
    // 'aurora-mutate' without any suffix
    if (!title) {
      return null;
    }

    return (
      <StoreItem
        key={item.package.name}
        title={removePrefix(item.package.name)}
        description={item.package.description}
        onClick={() => {
          onClick(item.package);
        }}
      />
    );
  });

  return <div>{List}</div>;
};

StoreItemList.propTypes = {
  items: PropTypes.array,
  onClick: PropTypes.func.isRequired
};

export default StoreItemList;
