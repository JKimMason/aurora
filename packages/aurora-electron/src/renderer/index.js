import React from "react";
import ReactDOM from "react-dom";
import Core from "../../../aurora-core/index.js";

export default () => {
  ReactDOM.render(<Core />, document.getElementById("root"));
};
