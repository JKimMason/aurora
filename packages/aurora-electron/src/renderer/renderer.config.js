/**
 * Webpack config for compiling renderer
 */

const path = require("path");

module.exports = {
  context: path.resolve(__dirname),
  entry: "./index.js",
  output: {
    path: path.resolve(".compiled"),
    filename: "renderer.bundle.js",
    library: "app",
    libraryTarget: "commonjs2"
  },
  target: "electron-renderer",
  module: {
    loaders: [
      {
        test: /\.js$/,
        loader: "babel-loader",
        exclude: /node_modules/,
        options: {
          presets: ["es2015", "react"]
        }
      },
      {
        test: /\.jsx$/,
        loader: "babel-loader",
        exclude: /node_modules/,
        options: {
          presets: ["es2015", "react"]
        }
      }
    ]
  }
};
