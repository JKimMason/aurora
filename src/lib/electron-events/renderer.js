import names from "./names";
import { ipcRenderer } from "electron";

/**
 * Senders
 */
function sendInstallMutation(name) {
  ipcRenderer.send(names.INSTALL_MUTATION, name);
}

/**
 * Listeners
 */
function onChangeScreen(then) {
  return ipcRenderer.on(names.CHANGE_SCREEN, then);
}

function onMutationInstalled(then) {
  return ipcRenderer.on(names.INSTALL_MUTATION_REPLY, then);
}

export default {
  onChangeScreen,
  onMutationInstalled,
  sendInstallMutation
};
