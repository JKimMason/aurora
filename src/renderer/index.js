import React from "react";
import ReactDOM from "react-dom";
import ScreenManager from "./ScreenManager";
import { Provider } from "react-redux";
import reducers from "../redux/reducers";
import { setPreferences } from "../redux/actions";
import { createStore } from "redux";
import { loadPreferences } from "../lib/io";
import Draggable from "../lib/ui/Draggable";

const store = createStore(reducers);

class Aurora extends React.Component {
  constructor(props) {
    super(props);

    loadPreferences().then(prefs => {
      store.dispatch(setPreferences(prefs));
    });
  }

  render() {
    return (
      <Provider store={store}>
        <Draggable dragRegionSize="50px">
          <ScreenManager />
        </Draggable>
      </Provider>
    );
  }
}

// Add Electron to the page
document.body.innerHTML = `<div id="root"></div>`;
ReactDOM.render(<Aurora />, document.getElementById("root"));
