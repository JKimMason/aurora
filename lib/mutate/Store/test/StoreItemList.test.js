import React from "react";
import StoreItemList from "../StoreItemList";
import renderer from "react-test-renderer";
import { withTheme } from "../../../theme";

const StoreItemListInContext = withTheme(StoreItemList);

describe("StoreItemList", () => {
  it("renders correctly with no items", () => {
    const dummyItems = [];
    const tree = renderer
      .create(<StoreItemListInContext items={dummyItems} onClick={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with items", () => {
    const dummyItems = [
      {
        package: {
          name: "one",
          description: "desc_one"
        }
      },
      {
        package: {
          name: "two",
          description: "desc_two"
        }
      }
    ];
    const tree = renderer
      .create(<StoreItemListInContext items={dummyItems} onClick={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
