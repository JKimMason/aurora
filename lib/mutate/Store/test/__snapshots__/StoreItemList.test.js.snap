// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreItemList renders correctly with items 1`] = `
<div
  className="sc-bdVaJa kDnqCF"
>
  <div>
    <div
      className="sc-iwsKbI lhbiTu"
    >
      <h2
        className="sc-dnqmqq hrMvoW"
      >
        one
      </h2>
      <p>
        desc_one
      </p>
      <button
        className="sc-bZQynM cEiCWc"
        onClick={[Function]}
      >
        Install
      </button>
    </div>
    <div
      className="sc-iwsKbI lhbiTu"
    >
      <h2
        className="sc-dnqmqq hrMvoW"
      >
        two
      </h2>
      <p>
        desc_two
      </p>
      <button
        className="sc-bZQynM cEiCWc"
        onClick={[Function]}
      >
        Install
      </button>
    </div>
  </div>
</div>
`;

exports[`StoreItemList renders correctly with no items 1`] = `
<div
  className="sc-bdVaJa kDnqCF"
>
  <div />
</div>
`;
